pennylane/operation.py

```diff
@@ -906,6 +906,50 @@ def __matmul__(self, other):
 
         raise ValueError("Can only perform tensor products between observables.")
 
+    def obs_data(self):
+        r"""Extracts the data from an Observable/Tensor."""
```

Comment:
Since this is a public method, I would expand the docstring to detail exactly what it is doing, how it is used, what it returns, etc.

Reply:
Agree, looking with a fresh pair of eyes it was hard to see what is happening.